libraries/app-iframe-message-react/src/useIframeChild.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  AppIFrameApplicationResponse,
+  AppIFramePlatformProcedures,
+  AppIFramePlatformRequest,
+  PlatformCrtIframeApi,
+  PlatformHandleRequestMap,
+} from "app-iframe-message/src";
+import { useCallback, useEffect, useMemo, useRef, useState } from "react";
+
+export const useIframeChild = (
+  handleRequest: PlatformHandleRequestMap,
+  onAppAvailable: (
+    iframe: HTMLIFrameElement,
+    api: PlatformCrtIframeApi,
+  ) => Promise<void>,
+): {
+  sendRequest: <ProcedureName extends AppIFramePlatformProcedures>(
+    request: Omit<AppIFramePlatformRequest, "id" | "type"> & {
+      procedure: ProcedureName;
+    },
+  ) => Promise<
+    AppIFrameApplicationResponse & {
+      procedure: ProcedureName;
+      type: "response";
+    }
+  >;
+  iframeRef: (node: HTMLIFrameElement | null) => void;
+} => {
+  const [iframe, setIFrame] = useState<HTMLIFrameElement | null>(null);
+  const isIFrameLoaded = useRef<boolean>(false);
+
+  const crtPlatform = useRef<PlatformCrtIframeApi>(
+    new PlatformCrtIframeApi(handleRequest),
+  );
```

Comment:
I do think the overall logic would become a lot simpler if we accept that the `crtPlatform` is part of a state, which only gets set once the various bits are available.

However, it'd require the hook to return `isReady, sendRequest, iframeRef` with the latter 2 being only defined if `isReady` is `true`, or something like that.

I'm somewhat concerned about `sendRequest` throwing in some hard-to-predict circumstances.

Reply:
Let's discuss this in person!